ui/ducks/send/send.js

```diff
@@ -1240,6 +1261,76 @@ export function signTransaction() {
   };
 }
 
+export function editTransaction(
+  assetType,
+  transactionId,
+  tokenData,
+  assetDetails,
+) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const unapprovedTransactions = getUnapprovedTxs(state);
+    const transaction = unapprovedTransactions[transactionId];
+    const { txParams } = transaction;
+    if (assetType === ASSET_TYPES.NATIVE) {
+      const {
+        from,
+        gas: gasLimit,
+        gasPrice,
+        to: address,
+        value: amount,
+      } = txParams;
+      const nickname = getAddressBookEntry(state, address)?.name ?? '';
+      await dispatch(
+        actions.editTransaction({
+          id: transactionId,
+          gasLimit,
+          gasPrice,
+          from,
+          amount,
+          address,
+          nickname,
+        }),
```

Comment:
I might be missing something but is there a reason we don't `updateSendAsset({type: ASSET_TYPES.NATIVE ...)}` here?

Reply:
Or actually I think it might be that the updateSendAsset call in the else block is redundant? From playing around with it a bit I can't think of or find a scenario where the assetType / details aren't already updated and aligned when you reach the updateSendAsset call on line 1313, could very well be missing something though!

Reply:
You have to set the asset on the send flow that you are going to be editing. The default send asset is native so that's why we don't call it in the first block.

Scenario 1: clicking edit from the send-eth-confirmation page
1. State is initialized with asset type = NATIVE
2. we call editTransaction which changes stage to EDIT
2. initialization of the send slice happens and sets the asset.balance to equal account.balance

Scenario 2: clicking edit from the send-token-confirmation page
1. State is initialized with asset type = TOKEN
2. We first update asset/balance which sets the appropriate values
3. We then call editTransaction which changes stage to EDIT
4. initilization occurs and uses the balance and asset we just set in state.

Reply:
Yup that checks out!